src/txn/txn_timestamp.c

```diff
@@ -463,6 +467,20 @@ __wt_txn_global_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
     WT_RET(__wt_txn_parse_timestamp(session, "oldest", &oldest_ts, &oldest_cval));
     WT_RET(__wt_txn_parse_timestamp(session, "stable", &stable_ts, &stable_cval));
 
+    /*
+     * The supplied oldest timestamp must be no older than the oldest timestamp in the history store
+     * checkpoint list. Note that the oldest history store timestamp is set during recovery and that
+     * if set, we reset it to WT_TS_NONE to ensure this code is only executed on a restart.
+     */
+    if (txn_global->oldest_ckpt_hs_timestamp > oldest_ts) {
+        txn_global->oldest_ckpt_hs_timestamp = WT_TS_NONE;
+        WT_RET_MSG(session, EINVAL,
+          "set_timestamp: oldest timestamp %s must not be earlier than oldest start timestamp %s "
```

Comment:
is it oldest start timestamp ? or oldest checkpoint timestamp?

Reply:
I updated the comment to clarify, thanks!